### ISSUE TITLE
file_readlink really working?

### ISSUE THREAD

**Issue by GillesDuvert** [gnudatalanguage/gdl#1709]
see https://github.com/gnudatalanguage/gdl/discussions/1708#discussioncomment-8021770

**Comment by alaingdl** [comment]
Unclear for me why the code is broken, if I change around line 2206 in file .cpp
```
       if( readlink(symlinkpath, actualpath, PATH_MAX) != -1)
	  actualpath[lenpath]='\0';
 
```
into
```
	int pid=readlink(symlinkpath, actualpath, PATH_MAX);
	cout << " pid " << actualpath << endl;
```

I got the right answer ... (doing ls -lart /proc/"pid"/ I can see it is the good one)

**Comment by alaingdl** [comment]
In the line after, `lenpath` is 0 (zéro), CQFD ! (QED)
then the logic around is wrong !

**Comment by alaingdl** [comment]
my snippet is wrong but I think I have solved the code and I have few tests in mind

**Comment by alaingdl** [comment]
Should be solved in PR #1713 
Please open a new issue if related problem
I did not had a lot of idea for tests ...
